### ISSUE TITLE
Expected trunk road is not chosen for Right Turn

### ISSUE THREAD

**Issue by alext951** [organicmaps/organicmaps#5997]
**Describe the issue**
Pls see the screenshot showing that the right turn access road is not selected by the routing algorithm. 
OSM id of the trunk road is: https://www.openstreetmap.org/way/11726960

What's more, the right turn suggested is actually not allowed, because both lanes of the main road are only going left. On OSM this is indicated by `turn:lanes	left|left` (see https://www.openstreetmap.org/way/61731320). 

Asking this on the Telegram chat, it appears that OM doesn't look at the `turn:lanes` tag at the moment, instead honoring the restriction relation.

Question here is two fold:
1. Should OM honor `turn:lanes`? In this case turn:lanes represents accurate road markings, and there are no road signs to warrant restriction relation. Or, alternatively, should restriction relation be added to OSM to reflect road markings (even though there is no road sign).
2. Organic Maps should route through the trunk road regardless of restrictions because that's just the natural route a driver would take. I am not sure though what is causing OM mis-route here. Implementing turn penalties has been mentioned on Telegram as a possible solution to this

**Expected behavior**
See point 2 above

**Screenshots**
![image](https://github.com/organicmaps/organicmaps/assets/43077263/a719a800-04d0-42ce-8408-f0a57743b9a3)

**System information:**
 - Operating system and its version: [Android 12]
 - Organic Maps version: 2023-08-18-8
 - Device Model: Pixel 5

**Comment by vng** [comment]
- In general, it is not obvious how to use turn:lanes tags for routing graph, because "left" can have multiple choices on complex transitions.
- Mappers should use restriction relations here.

- In this particular case OM should take into account traffic lights and increase a penalty of a 90-degree right turn. And use traffic-free link instead.

BTW, GraphHopper makes the same error: https://www.openstreetmap.org/directions?engine=graphhopper_car&route=40.94901%2C-74.27446%3B40.94972%2C-74.27619#map=19/40.94927/-74.27533

**Comment by alext951** [comment]
Yes @vng - thank you for outlining these points, makes complete sense. Penalty for a 90-degree turn (and maybe traffic lights) sounds like the right solution to this specific issue rather than turn:lane tags.
